src/platform/src/main/java/org/geoserver/platform/resource/Paths.java

```diff
@@ -423,7 +423,12 @@ public static String convert(String path, String filename) {
      */
     public static File toFile(File base, String path) {
         for (String item : Paths.names(path)) {
-            base = new File(base, item);
+            base =
+                    new File(
+                            base,
+                            base == null && File.separator.equalsIgnoreCase("\\")
+                                    ? item + "\\"
+                                    : item);
```

Comment:
Do you have a test case to show what you are  does? You can make a windows only test case and it will be picked up by the GitHub actions.

We are explicitly using the file constructors here rather than ``/`` or ``\\`` to avoid system dependencies so I am a little startled by your suggested change.

But I see your point, the javadoc is clear that "in the case of absolute path base should be null ... 

- Is the path you are thinking of using the form ``C:\\path\filename.txt`` ?
- Network shares should be of the form ``\\\\machine\\dirve\path\filename.ext``

Consider checking File.listRoots() - to check first item, seems complicated.

Reply:
ok that's a good suggestion; I realized that this did not show up ususally and I wondered why it didn't when the system is tested under windows github action, and I definitely didn't consider the network share situation.  I'll try again.  Thanks!

Reply:
Note that there is some post-Paths.toFile work going on [here ](https://github.com/geoserver/geoserver/blob/770dc6f7023bc2ab32597cfc7a3a9cc35ff3b608/src/platform/src/test/java/org/geoserver/platform/resource/FileWrapperResourceTheoryTest.java#L32-L33) that is addressing on linux the problem that is encountered on Windows, namely an absolute path appearing as relative after toFile(null, path).  I pushed a test that fails on linux and unfortunately I can't deal with it until tonight.

Reply:
@jodygarnett  So I set up two branches that solve the problem (on Windows, Linux TBD).  The first branch is the same approach taken here, with hopefully the linux test failure removed, [here](https://github.com/geoserver/geoserver/compare/main...Maps4HTML:GEOS-9949-A).

The second approach leaves the Paths.toFile method unchanged, and applies some compensating gyrations to where the failure occurs (in FileWrapperResourceTheoryTest.getResource method) the branch is [here](https://github.com/geoserver/geoserver/compare/main...Maps4HTML:GEOS-9949-B).

Neither approach is great, even assuming they work on both systems.  I'm starting to think all this is to work around a bug in java...File that doesn't allow it to correctly resolve paths in the constructor. 

Please let me know what you think.  No rush, I realize you have a ton on your plate right now. Thanks!